### PR TITLE
doc: replace .known-issues with warnings_filter

### PR DESCRIPTION
**Pull request by gmarull**
Summary:

```
- doc: nrf: enable warnings_filter
Use warnings_filter extension to filter nrf warnings.

- doc: remove known-issues filter mechanism
Remove the "known-issues" mechanism used to filter the known warnings in
our docs. CI will build now with "-W", that is, treat warnings as
errors.

Note: intermediate builds such as *-inventory do not take extra Sphinx
options (e.g. -W on CI) since they likely have cross-references that are
impossible to resolve on a first-pass build.

- doc: scripts: remove old filter-known-issues script
Script is not needed anymore.

- doc: nrf: document how to filter warnings
Add some details on how warnings filtering works.
```

### CODE REVIEW

**Comment by gmarull** [comment]
note: Jenkins failures are not related to these changes

File: .github/workflows/docbuild.yml [hunk 1]
```diff
@@ -58,39 +58,9 @@ jobs:
         run: |
           export PATH="$HOME/.local/bin:$PATH"
           mkdir -p _build && cd _build
-          cmake -GNinja ../nrf/doc
+          cmake -GNinja ../nrf/doc -DEXTRA_SPHINXOPTS="-W"
           ninja build-all
 
-      - name: Check build warnings
```
Comment (carlescufi) [inline .github/workflows/docbuild.yml:64]: Where are you checking if there are warnings now? Just the ninja exit code?

File: scripts/sphinx_extensions/warnings_filter.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
+"""
```
Comment (carlescufi) [inline scripts/sphinx_extensions/warnings_filter.py:1]: Why can't we use the upstream one directly?

File: .github/workflows/docbuild.yml (hunk 1 above)
Comment (gmarull) [reply .github/workflows/docbuild.yml:64]: Return code will be non-zero:
```
Warning, treated as error:
/.../ncs/nrf/doc/_build/nrf/rst/doc/nrf/include/bluetooth/mesh/light_ctl_cli.rst:35:Duplicate C declaration, also defined at include/bluetooth/mesh/light_ctl:35.
Declaration is '.. c:function:: int bt_mesh_light_ctl_set (struct bt_mesh_light_ctl_cli *cli, struct bt_mesh_msg_ctx *ctx, const struct bt_mesh_light_ctl_set *set, struct bt_mesh_light_ctl_status *rsp)'.
ninja: build stopped: subcommand failed.
ninja: build stopped: subcommand failed.
(.venv) ➜  nrf git:(feature/remove-known-issues) ✗ echo $?
1
```

File: scripts/sphinx_extensions/warnings_filter.py (hunk 2 above)
Comment (gmarull) [reply scripts/sphinx_extensions/warnings_filter.py:1]: if we want to be self-contained w.r.t our docset extensions it's better to have our own set, I'm fine using the upstream one though.

File: .github/workflows/docbuild.yml (hunk 1 above)
Comment (ru-fu) [reply .github/workflows/docbuild.yml:64]: So if we have an error in the Zephyr build (locally), we cannot use build-all?

And we don't have any files anymore that contain the errors?

Comment (gmarull) [reply .github/workflows/docbuild.yml:64]: Yes, we can. `-W` option is only enforced on CI.

Comment (gmarull) [reply .github/workflows/docbuild.yml:64]: regarding errors, see the `doc/nrf/known-warnings.txt` and the updated docs detailing the mechanism.

File: scripts/sphinx_extensions/warnings_filter.py (hunk 2 above)
Comment (gmarull) [reply scripts/sphinx_extensions/warnings_filter.py]: using upstream for now

**Review by carlescufi — APPROVED** [review]

**Review by ru-fu — CHANGES_REQUESTED** [review]
I don't have permission to push to your branch or make a PR, but please include the following update: https://github.com/ru-fu/sdk-nrf/commit/18c5cb6a2e62e90293dabe83705a20fef245dc92

File: doc/nrf/doc_build.rst [hunk 3]
```diff
@@ -216,3 +216,18 @@ To test the version drop-down locally, complete the following steps:
...
+
+However, there are some known issues with Sphinx and Breathe that generate Sphinx warnings even though the input is valid C code.
+To deal with such unavoidable warnings, Zephyr provides the Sphinx extension ``zephyr.warnings_filter`` that filters out warnings based on a set of regular expressions.
+You can find the extension at :file:`ncs/zephyr/doc/_extensions/zephyr/warnings_filter.py`.
+
+The extension has to be enabled only on docsets that require it. It can be done in the ``conf.py`` file of each docset.
+The extension relies on an external file containing regular expressions to filter out warnings related to duplicate C declarations.
+These warnings are caused by different objects (for example, a struct and a function or nested elements) sharing the same name.
+The regular expressions file should be located under each docset folder.
+For example, in case of ``nrf`` it can be found at :file:`ncs/nrf/doc/nrf/known-warnings.txt`.
```
Comment (ru-fu) [inline doc/nrf/doc_build.rst:233]:
```suggestion
The configuration file that defines the expected warnings for the nrf documentation set is located at :file:`ncs/nrf/doc/nrf/known-warnings.txt`.
It contains regular expressions to filter out warnings related to duplicate C declarations.
These warnings are caused by different objects (for example, a struct and a function or nested elements) sharing the same name.
```

Comment (gmarull) [reply doc/nrf/doc_build.rst:233]: This change removes the details on how to enable warnings in other docsets other than `nrf`, this information needs to be provided. Other future docsets (e.g. nrfx) will also need this feature enabled.

Comment (gmarull) [reply doc/nrf/doc_build.rst]: adjusted

**Review by ru-fu — APPROVED** [review]

**Pull request merged**